d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -331,6 +360,16 @@ private void readyHandler()
   void handleResponse(DiscoveryResponseData response)
   {
     ResourceType resourceType = response.getResourceType();
+    // capturing the received resource versions, which will be used to set IRV for re-connect scenarios.
+    Map<String, String> resourceVersions = getResourceVersions().get(resourceType);
+    for (Resource res: response.getResourcesList()){
+      resourceVersions.put(res.getName(), res.getVersion());
+    }
+
+    for (String removedResource: response.getRemovedResources()){
+      resourceVersions.remove(removedResource);
+    }
```

Comment:
Watch the formatting here, should respect the local style:
```suggestion
    // capturing the received resource versions, which will be used to set IRV for re-connect scenarios.
    Map<String, String> resourceVersions = getResourceVersions().get(resourceType);
    for (Resource res: response.getResourcesList())
    {
      resourceVersions.put(res.getName(), res.getVersion());
    }

    for (String removedResource: response.getRemovedResources())
    {
      resourceVersions.remove(removedResource);
    }
```